plugins/catalog/src/api/CatalogClient.ts

```diff
@@ -42,4 +43,23 @@ export class CatalogClient implements CatalogApi {
     if (entity) return entity;
     throw new Error(`'Entity not found: ${name}`);
   }
+  async getLocationByEntity(entity: Entity): Promise<any> {
+    const findLocationIdInEntity = (e: Entity) => {
+      return e.metadata.annotations
+        ? e.metadata.annotations['backstage.io/managed-by-location']
```

Comment:
you can just do `const locationId = e.metadata.annotations?.['backstage.io/managed-by-location']`, then it'll be of type `string | undefined`